### DISCUSSION TITLE
After updating, knocks out an error during write-off operations

### DISCUSSION THREAD

**Discussion by nloshka** [firefly-iii/firefly-iii#4403] [Firefly III]
Debug information generated at 2021-02-16 14:08:28 Europe/Amsterdam for Firefly III version **5.5.0-beta.1**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.5.0-beta.1 |
| Firefly III API | 1.5.0 |
| PHP | 7.4.12 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Installation ID | 28629bf3-d0fb-404b-a873-af978b1c355d |
| Using docker? | false |
| Telemetry | false |
| Layout |  |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, daily |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | fpm-fcgi |
| Default language | en_US |
| Default locale | equal |
| BCscale | 12 |
| DB drivers | mysql, sqlite |
| Current driver | mysql |
| DB version | 15 (exp. 15) |
| Login provider | eloquent  |
| Trusted proxies (.env) |  |

| User info | Value |
| --- | --- |
| Session start | 2021-02-01 00:00:00 |
| Session end | 2021-02-28 23:59:59 |
| Session first | 2018-12-08 00:00:00 |
| User ID | 1 |
| User language | ru_RU |
| User locale | ru_RU |
| Attempt at "ru_RU.utf8" | 'ru_RU.utf8' |
| Attempt at "ru_RU.UTF-8" | 'ru_RU.UTF-8' |
| User agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36 |
How do I resolve the error?
Mistake:Internal Firefly III Exception: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'buh.webhook_messages' doesn't exist (SQL: select `webhook_messages`.* from `webhook_messages` where `webhook_messages`.`sent` = 0 and `webhook_messages`.`errored` = 0)

**Comment by JC5** [comment]
Make sure you run the migrations after updating.

**Comment by nloshka** [comment]
how to do migration? what commands do you need to execute? Thank.

**Comment by sephrat** [comment]
https://docs.firefly-iii.org/firefly-iii/advanced-installation/upgrade/

**Comment by nloshka** [comment]
# php artisan firefly-iii:upgrade-database
Now seeding the database...

In Connection.php line 678:
                                                                                                                                    
  SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'failed_jobs' already exists (SQL: create table `failed_jobs` (`i  
  d` bigint unsigned not null auto_increment primary key, `uuid` varchar(191) not null, `connection` text not null, `queue` text n  
  ot null, `payload` longtext not null, `exception` longtext not null, `failed_at` timestamp default CURRENT_TIMESTAMP not null) d  
  efault character set utf8mb4 collate 'utf8mb4_unicode_ci' engine = InnoDB)                                                        
                                                                                                                                    

In Connection.php line 471:
                                                                                               
  SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'failed_jobs' already exists

**Comment by nloshka** [comment]
# php artisan migrate --seed
Migrating: 2020_11_12_070604_changes_for_v550

In Connection.php line 678:
                                                                                                                                    
  SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'failed_jobs' already exists (SQL: create table `failed_jobs` (`i  
  d` bigint unsigned not null auto_increment primary key, `uuid` varchar(191) not null, `connection` text not null, `queue` text n  
  ot null, `payload` longtext not null, `exception` longtext not null, `failed_at` timestamp default CURRENT_TIMESTAMP not null) d  
  efault character set utf8mb4 collate 'utf8mb4_unicode_ci' engine = InnoDB)                                                        
                                                                                                                                    

In Connection.php line 471:
                                                                                               
  SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'failed_jobs' already exists

**Comment by JC5** [comment]
Tell us how you installed the beta?

**Reply by nloshka** [reply]
The selected update channel is stable, but why is the beta version installed? How to fix it?

**Reply by JC5** [reply]
How did you install Firefly III in the first place?

**Reply by nloshka** [reply]
git pull
rm -rf bootstrap/cache/*
rm -rf vendor/
composer install --no-scripts --no-dev
composer install --no-dev
php artisan migrate --seed
php artisan firefly-iii:decrypt-all
php artisan cache:clear
php artisan firefly-iii:upgrade-database
php artisan passport:install
php artisan cache:clear



always through the git and up to this point everything worked fine and was updated.

**Reply by JC5** [reply]
The channel is stable but that's just the update check, not the code. Pulling code from the main branch will have pulled in the beta.

What's the content of the migrations table?

**Reply by nloshka** [reply]
How to view? In the database? or in a file somewhere?

**Reply by nloshka** [reply]
-- phpMyAdmin SQL Dump
-- version 5.0.2
-- https://www.phpmyadmin.net/
--
-- Хост: localhost
-- Время создания: Фев 17 2021 г., 08:47
-- Версия сервера: 5.5.68-MariaDB
-- Версия PHP: 7.4.12

SET SQL_MODE = "NO_AUTO_VALUE_ON_ZERO";
START TRANSACTION;
SET time_zone = "+00:00";


/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!40101 SET @OLD_CHARACTER_SET_RESULTS=@@CHARACTER_SET_RESULTS */;
/*!40101 SET @OLD_COLLATION_CONNECTION=@@COLLATION_CONNECTION */;
/*!40101 SET NAMES utf8mb4 */;

--
-- База данных: `buh`
--

-- --------------------------------------------------------

--
-- Структура таблицы `migrations`
--

CREATE TABLE `migrations` (
  `id` int(10) UNSIGNED NOT NULL,
  `migration` varchar(191) COLLATE utf8mb4_unicode_ci NOT NULL,
  `batch` int(11) NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;

--
-- Дамп данных таблицы `migrations`
--

INSERT INTO `migrations` (`id`, `migration`, `batch`) VALUES
(1, '2016_06_16_000000_create_support_tables', 1),
(2, '2016_06_16_000001_create_users_table', 1),
(3, '2016_06_16_000002_create_main_tables', 1),
(4, '2016_08_25_091522_changes_for_3101', 1),
(5, '2016_09_12_121359_fix_nullables', 1),
(6, '2016_10_09_150037_expand_transactions_table', 1),
(7, '2016_10_22_075804_changes_for_v410', 1),
(8, '2016_11_24_210552_changes_for_v420', 1),
(9, '2016_12_22_150431_changes_for_v430', 1),
(10, '2016_12_28_203205_changes_for_v431', 1),
(11, '2017_04_13_163623_changes_for_v440', 1),
(12, '2017_06_02_105232_changes_for_v450', 1),
(13, '2017_08_20_062014_changes_for_v470', 1),
(14, '2017_11_04_170844_changes_for_v470a', 1),
(15, '2018_01_01_000001_create_oauth_auth_codes_table', 1),
(16, '2018_01_01_000002_create_oauth_access_tokens_table', 1),
(17, '2018_01_01_000003_create_oauth_refresh_tokens_table', 1),
(18, '2018_01_01_000004_create_oauth_clients_table', 1),
(19, '2018_01_01_000005_create_oauth_personal_access_clients_table', 1),
(20, '2018_03_19_141348_changes_for_v472', 1),
(21, '2018_04_07_210913_changes_for_v473', 1),
(22, '2018_04_29_174524_changes_for_v474', 1),
(23, '2018_06_08_200526_changes_for_v475', 1),
(24, '2018_09_05_195147_changes_for_v477', 2),
(25, '2018_11_06_172532_changes_for_v479', 2),
(26, '2019_01_28_193833_changes_for_v4710', 3),
(27, '2019_02_05_055516_changes_for_v4711', 4),
(28, '2019_02_11_170529_changes_for_v4712', 5),
(29, '2019_03_11_223700_fix_ldap_configuration', 6),
(30, '2019_03_22_183214_changes_for_v480', 7),
(31, '2019_11_30_000000_create_2fa_token_table', 8),
(32, '2019_12_28_191351_make_locations_table', 9),
(33, '2020_03_13_201950_changes_for_v520', 10),
(34, '2020_06_07_063612_changes_for_v530', 11),
(35, '2020_06_30_202620_changes_for_v530a', 11),
(36, '2020_07_24_162820_changes_for_v540', 11);

--
-- Индексы сохранённых таблиц
--

--
-- Индексы таблицы `migrations`
--
ALTER TABLE `migrations`
  ADD PRIMARY KEY (`id`);

--
-- AUTO_INCREMENT для сохранённых таблиц
--

--
-- AUTO_INCREMENT для таблицы `migrations`
--
ALTER TABLE `migrations`
  MODIFY `id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT, AUTO_INCREMENT=37;
COMMIT;

/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;
/*!40101 SET CHARACTER_SET_RESULTS=@OLD_CHARACTER_SET_RESULTS */;
/*!40101 SET COLLATION_CONNECTION=@OLD_COLLATION_CONNECTION */;

**Reply by JC5** [reply]
Thanks. Delete the "failed_jobs" table from the database then run the commands again.

**Reply by nloshka** [reply]
# php artisan firefly-iii:upgrade-database
Now seeding the database...

In Connection.php line 678:
                                                                                                                                         
  SQLSTATE[42S21]: Column already exists: 1060 Duplicate column name 'budget_limit_id' (SQL: alter table `budget_transaction_journal` a  
  dd `budget_limit_id` int unsigned null after `budget_id`)                                                                              
                                                                                                                                         

In Connection.php line 471:
                                                                                        
  SQLSTATE[42S21]: Column already exists: 1060 Duplicate column name 'budget_limit_id'

**Reply by JC5** [reply]
Sorry, I'm out of ideas. It looks like half your database is updated and the other half isn't. I can't fix that remotely. :(

**Reply by nloshka** [reply]
But why then did the beta channel work instead of prod? How can I fix this can there be an instruction? or any thoughts on this?
Thank.

**Reply by JC5** [reply]
The stable channel only tells you which updates are available for your channel. `git pull` doesn't work with that. `git pull` will download whatever version is there (stable, beta and alpha).

**Reply by JC5** [reply]
I fear you must grab your backup and go back to 5.4.6.

**Reply by nloshka** [reply]
Thanks I got it. but since the last update, a lot has already been made and it is no longer possible to make some new accounts. I will use the beta version, I hope there is nothing wrong with that.

**Reply by nloshka** [reply]
If I send you a dump of the database, you can fix the error without losing data from the tables:
Internal Firefly III Exception: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'buh.webhook_messages' doesn't exist (SQL: select `webhook_messages`.* from `webhook_messages` where `webhook_messages`.`sent` = 0 and `webhook_messages`.`errored` = 0)

**Reply by nloshka** [reply]
> I fear you must grab your backup and go back to 5.4.6.

Ping...
If I send you a dump of the database, you can fix the error without losing data from the tables:
Internal Firefly III Exception: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'buh.webhook_messages' doesn't exist (SQL: select webhook_messages.* from webhook_messages where webhook_messages.sent = 0 and webhook_messages.errored = 0)

**Reply by nloshka** [reply]
Or maybe I can do it myself? Just tell me what to do and I'll do it.
Thank.
Excuse for troubling.
